build/mixxx.py

```diff
@@ -213,15 +213,15 @@ def __init__(self, target, machine, build, toolchain, available_features):
         # Now that environment variables have been read, we can detect the compiler.
         import subprocess
         process = subprocess.Popen("%s %s" %(self.env['CC'], '--version'), stdout=subprocess.PIPE, shell=True) # nosec
-        (stdout, stderr) = process.communicate()
+        stdout = process.communicate()[0].rstrip().decode()
```

Comment:
Might make sense to use [`subprocess.check_output()`](https://docs.python.org/2/library/subprocess.html#subprocess.check_output) instead.

Reply:
Thanks for the hint!

Reply:
I just read that process.run() is recommended instead of subprocess.check_output() since 3.5? I'm not able to decide what to choose here, considering that we need to support a variety of Python versions. Maybe someone else can take over and fix all those issues?

Reply:
Since you only want to check the content of stdout anyway, there's no actual advantage from using `subprocess.run()` other than being hip and modern. On the other hand, we'd lose compatibility with all Python versions < 3.5 (`subprocess.check_output` even works on Python 2.7).